### ISSUE TITLE
Weak typing when pass an object as a key

### ISSUE THREAD

**Issue by mrfratello** [nanostores/query#45]
The type inference for the `filters` parameter in the `$staffsData` query is incorrect. It is inferred as `SomeKey` instead of the expected type `RequestFiltersDTO`.

Steps to Reproduce:

Define `$formFilters` atom and `$filters` computed:

```ts
export const $formFilters = atom<FormValue>({
  date: [dayjs().subtract(1, 'month'), dayjs()]
});

export const $filters = computed($formFilters, ({ date, ...formValues }) => {
  const payload: RequestFiltersDTO = formValues;
  if (date) {
    payload.date_from = date[0].startOf('day').toISOString();
    payload.date_to = date[1].endOf('day').toISOString();
  }
  return payload;
});
```

In this case, the `$filters` should be of type `RequestFiltersDTO`. 

However, when examining the definition of the `$staffData` query, we encounter a TypeScript error:

```ts
const [query] = nanoquery({});

export const $staffsData = query([$filters], { // <-- TypeScript error here
  fetcher: (filters) => getAwesomeRequest(filters), // <-- `filters` is inferred as `SomeKey` instead of `RequestFiltersDTO`
});
```

Here, the `$staffsData` query has `$filters` defined as a dependency. However, the `filters` parameter in the fetcher function has an incorrect type inference.

**Comment by dkzlv** [comment]
@mrfratello Hey there!

This is not a bug, but rather a design decision. More [on that here in the doc](https://github.com/nanostores/query?tab=readme-ov-file#createfetcherstore) and [in this comment](https://github.com/nanostores/query/issues/7#issuecomment-1523208109).

TLDR: values you provide in the dependencies get serialized into strings to act as cache keys. I've seen people pass the whole entity (say, a post or a comment) into SWR that makes the cache so much bigger and the app less efficient that I decided we shouldn't allow this and stick to primitives instead (strings, numbers, bools). You can circumvent this limitation by using `JSON.stringify`/`parse`, or by changing the model layer somehow.

Lmk if something's missing in the explanation!

**Comment by mrfratello** [comment]
Understood, thanks.

I have reimplemented the code as follows:

```ts
const $filtersKey = computed($filters, JSON.stringify);

export const $staffsData = query([$filtersKey], {
  fetcher: () => {
    const filters = $filters.get();
    return getAwesomeRequest(filters);
  },
});
```
